src/haz3lschool/SyntaxTest.re

```diff
@@ -0,0 +1,237 @@
+open Haz3lcore;
+open Sexplib.Std;
+
+module StringMap = Map.Make(String);
+
+[@deriving (show({with_path: false}), sexp, yojson)]
+type syntax_tests = {
```

Comment:
this should support an arbitrary list of syntax tests, not just these two examples -- any suitable predicate on the syntax is a valid syntax test